python/lsst/meas/deblender/plugins.py

```diff
@@ -222,11 +222,9 @@ def buildMultibandTemplates(debResult, log, useWeights=False, usePsf=False,
     fpMask = afwImage.Mask(bbox)
     debResult.footprint.spans.setMask(fpMask, 1)
     fpMask = ~fpMask.getArray().astype(bool)
-    mask = np.zeros(weights.shape, dtype=bool)
-    for fidx, mimg in enumerate(maskedImages):
-        badPixels = mimg.mask.getPlaneBitMask(badMask)
-        mask[fidx] = (mimg.getMask().array & badPixels) | fpMask
-    weights[mask] = 0
+    badPixels = mMaskedImage.mask.getPlaneBitMask(badMask)
+    mask = (mMaskedImage.mask.array & badPixels) | fpMask[None, :]
+    weights[mask>0] = 0
```

Comment:
Was this intentional? Is it actually faster?

Reply:
I don't know if it's faster, but it's possible now and I think looks cleaner than the previous version.

Reply:
`weights[mask>0] ` is against standards anyway. Need whitespace around `>`.